docs/spec/spec.md

```diff
@@ -306,27 +306,27 @@ The following API writes given CSV content to a given file. When writing a `Reco
 # io:Error? resultRecord = io:fileWriteCsv("./resources/myfileRecord.csv", contentRecord);
 # ```
 # + path - The CSV file path
-# + content - CSV content as an array of string arrays
+# + content - CSV content as an array of string arrays or a array of ballerina records.
 # + option - To indicate whether to overwrite or append the given content
 # + return - An `io:Error` or `()` when the writing was successful
 public isolated function fileWriteCsv(string path, string[][] content, FileWriteOption option = OVERWRITE) returns
 Error?;
 ```
 
-The following API reads the content of a given CSV file as a stream of string arrays. Here, each CSV record represents as a string array.
+The following API reads the content of a given CSV file as a stream of string arrays or ballerina records. Here, each CSV record represents as a string array or a ballerina record.
 
 ```ballerina
 # Read file content as a CSV.
 # ```ballerina
 # stream<string[], io:Error?>|io:Error content = io:fileReadCsvAsStream("./resources/myfile.csv");
-# stream<map<anydata>, io:Error?>|io:Error content = io:fileReadCsvAsStream("./resources/myfile.csv");
+# stream<record{}, io:Error?>|io:Error content = io:fileReadCsvAsStream("./resources/myfile.csv");
 # ```
 # + path - The CSV file path
-# + return - The entire CSV content in the channel a stream of string arrays or an `io:Error`
+# + return - The entire CSV content in the channel a stream of string arrays, ballerina records or an `io:Error`
 public isolated function fileReadCsvAsStream(string path) returns stream<string[], Error?>|Error;
 ```
 
-The following API writes a given CSV stream to a given file.
+The following API writes a given CSV stream to a given file. When writing a `Record[]` content to a CSV file in `OVERWRITE`, by default, headers will be written to the CSV file as unlike `string[]` the order of record fields is not guaranteed. For `APPEND`, order of the existing csv file is inferred using the headers and used as the order
```

Comment:
```suggestion
The following API writes a given CSV stream to a given file. When writing string[] to CSV, the order of the array elements are considered. However when writing a `record{}[]`, the order is inferred from the CSV headers (CSV header should match the fields). Moreover, in OVERWRITE mode fields are written as headers to preserve order between writes.
```

Reply:
Lets add similar message in API docs. Spec is not the first place a user refers this.

Reply:
@sahanHe did we update the API Docs